src/openvic-simulation/military/UnitInstance.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "openvic-simulation/map/Province.hpp"
+#include "openvic-simulation/military/Leader.hpp"
+#include "openvic-simulation/military/UnitType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include <string_view>
+#include <type_traits>
+#include <vector>
+
+namespace OpenVic {
+	template<typename T, typename std::enable_if<std::is_base_of<UnitType, T>::value>::type* = nullptr>
+	struct UnitInstance {
+	protected:
+		std::string PROPERTY_RW(unit_name);
+		T const& PROPERTY(unit_type); //can't change
+
+		fixed_point_t PROPERTY_RW(organisation);
+		fixed_point_t PROPERTY_RW(morale);
+		fixed_point_t PROPERTY_RW(strength);
+
+	protected:
+		UnitInstance(std::string_view new_unit_name, T const& new_unit_type) : 
+			unit_name { new_unit_name },
+			unit_type { new_unit_type }, 
+			organisation { new_unit_type.get_default_organisation() }, //TODO: modifiers
+			morale { 0 }, //TODO: where
+			strength { new_unit_type.get_max_strength() } {}
+	};
+
+	struct RegimentInstance : UnitInstance<RegimentType> {
+	private:
+		Pop const& PROPERTY(pop); // can't change
+
+	public:
+		RegimentInstance(std::string_view new_name, RegimentType const& new_regiment_type, Pop const& new_pop);
+	};
+
+	struct ShipInstance : UnitInstance<ShipType> {
+	public:
+		ShipInstance(std::string_view new_name, ShipType const& new_ship_type);
+	};
+
+	struct MovementInfo {
+	private:
+		std::vector<Province const*> PROPERTY_RW(path);
+		fixed_point_t PROPERTY_RW(movement_progress);
+		bool PROPERTY_RW(locked);
+
+	public:
+		MovementInfo(Province const* starting_province, Province const* target_province); // contains/calls pathfinding logic
+	};
+
+	template<typename T, typename I, typename std::enable_if<std::is_base_of<UnitInstance<T>, I>::value>::type* = nullptr>
+	struct UnitInstanceGroup {
+	private:
+		std::string PROPERTY_RW(name);
+		const UnitType::branch_t PROPERTY(branch);
+		std::vector<I> PROPERTY(units);
+		Leader const* PROPERTY_RW(leader);
+		Province const* PROPERTY_RW(position);
+
+		std::optional<MovementInfo> PROPERTY_RW(movement_info);
+
+	protected:
+		UnitInstanceGroup(
+			std::string_view new_name,
+			UnitType::branch_t new_branch,
+			std::vector<I> new_units,
+			Leader const* new_leader,
+			Province const* new_position
+		) : name { new_name },
+			branch { new_branch },
+			units { new_units },
+			leader { new_leader },
+			position { new_position } {}
+	};
+
+	struct ArmyInstance : UnitInstanceGroup<RegimentType, RegimentInstance> {
```

Comment:
```suggestion
	struct ArmyInstance : UnitInstanceGroup<RegimentInstance> {
```
With `UnitInstanceGroup` using `utility::is_derived_from_specialization_of`, we don't need the first template parameter anymore (same for `NavyInstance`).